components/node-labeler/cmd/run.go

```diff
@@ -299,3 +305,62 @@ func waitForTCPPortToBeReachable(host string, port string, timeout time.Duration
 		}
 	}
 }
+
+func waitForRegistryFacade(host, port string, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	transport := newDefaultTransport()
+	transport.TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
+	client := &http.Client{
+		Transport: transport,
+	}
+
+	dummyURL := fmt.Sprintf("https://%v:%v/v2/remote/not-a-valid-image/manifests/latest", host, port)
+
+	for {
```

Comment:
instead of blocking the reconciliation, could we return and requeue an event for the reconciler? That way we don't fill up the amount of concurrent reconciliations and potentially limiting how many nodes we can handle

Reply:
Good point. 
done